src/main/java/igrad/ui/AvatarSelectionPanel.java

```diff
@@ -0,0 +1,70 @@
+package igrad.ui;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import igrad.commons.core.LogsCenter;
+import igrad.model.avatar.Avatar;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Region;
+import javafx.scene.text.Font;
+
+/**
+ * Panel containing the list of modules.
+ */
+public class AvatarSelectionPanel extends UiPart<Region> {
+    private static final String FXML = "AvatarSelectionPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(AvatarSelectionPanel.class);
+
+    @FXML
+    private GridPane avatarGridPane;
+
+    @FXML
+    private Label avatarLabel;
+
+    private List<AvatarImage> avatarImgList = new ArrayList<>();
+
+    public AvatarSelectionPanel() {
+        super(FXML);
+
+        showMainLabel();
```

Comment:
order should be swapped, `initAvatarImgList()` (internal operation) done before `showMainLabel()` (external operation) to avoid visible lag to user